cores/esp8266/core_esp8266_waveform.cpp

```diff
@@ -259,24 +527,58 @@ static ICACHE_RAM_ATTR void timer1Interrupt() {
         // Check for toggles
         int32_t cyclesToGo = wave->nextServiceCycle - now;
         if (cyclesToGo < 0) {
-          waveformState ^= mask;
-          if (waveformState & mask) {
+          uint32_t nextEdgeCycles;
+          uint32_t desired = 0;
+          uint32_t *timeToUpdate;
+          wvfState.waveformState ^= mask;
+          if (wvfState.waveformState & mask) {
             if (i == 16) {
-              GP16O |= 1; // GPIO16 write slow as it's RMW
+              GP16O = 1; // GPIO16 write slow as it's RMW
```

Comment:
With revision control, please never leave commented code or outdated comments in commits.